### PR TITLE
Reserve default keyword

### PR DESCRIPTION
**Pull request by bmc-msft**
Add `default` to the keywords that are ignored.

Regenerated services based on https://github.com/Azure/azure-rest-api-specs/commit/02fce64677f78021ba70d69bb8bd0d916d653927

### CODE REVIEW

File: services/autorust/codegen/src/identifier.rs [hunk 1]
```diff
@@ -174,6 +174,8 @@ fn is_keyword(word: &str) -> bool {
             | "where"
             | "while"
             | "yield"
+        // required for Default::default support
```
Comment (cataggar) [inline services/autorust/codegen/src/identifier.rs:177]: I don't understand why this change is required. What is not working?

**Review by cataggar — CHANGES_REQUESTED** [review]
Can you please provide an example?

File: services/autorust/codegen/src/identifier.rs (hunk 1 above)
Comment (bmc-msft) [reply services/autorust/codegen/src/identifier.rs:177]: This removes the confusion between `::default()` and `.default`, which have wildly different meanings.  Consider `ResourceSkuCapacity::default()` verses `ResourceSkuCapacity.default`:

https://github.com/Azure/azure-sdk-for-rust/pull/602/files#diff-4169133b67bc7ca3e13582d656b1ece2bae7ce37d3be10b0c6d0076fdedc242f

File: services/autorust/codegen/examples/gen_mgmt.rs [hunk 2]
```diff
@@ -285,6 +285,7 @@ const BOX_PROPERTIES: &[(&str, &str, &str)] = &[
     ("../../../azure-rest-api-specs/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/managedHsm.json", "Error" , "innererror"),
     ("../../../azure-rest-api-specs/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json", "Error" , "innererror"),
     ("../../../azure-rest-api-specs/specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/managedHsm.json", "Error" , "innererror"),
+    ("../../../azure-rest-api-specs/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2021-10-01/managedHsm.json", "Error", "innererror"),
```
Comment (cataggar) [inline services/autorust/codegen/examples/gen_mgmt.rs:288]: note for later. will need this ☝️

**Comment by cataggar** [comment]
Thanks for the proposal. I've moved it to issue #604 for now.
